File: app/src/main/java/com/elegion/android/template/ui/base/fragment/BaseRefresherFragment.java
package com.elegion.android.template.ui.base.fragment;

import android.support.v4.widget.SwipeRefreshLayout;

import com.elegion.android.template.R;

import butterknife.BindView;
import timber.log.Timber;

/**
 * @author Mike
 */
public abstract class BaseRefresherFragment extends BaseNoInternetFragment
        implements SwipeRefreshLayout.OnRefreshListener {
    @BindView(R.id.refresher)
    SwipeRefreshLayout mSwipeRefreshLayout;

    @Override
    public void onResume() {
        super.onResume();
        mSwipeRefreshLayout.setOnRefreshListener(this);
    }

    @Override
    public void onPause() {
        mSwipeRefreshLayout.setOnRefreshListener(null);
        super.onPause();
    }

    @Override
    public void showLoadingIndicator() {
        Timber.d("MOXY: Refresher showLoadingIndicator()");
        if (mSwipeRefreshLayout != null) {
            mSwipeRefreshLayout.post(() -> {
                if (mSwipeRefreshLayout != null) {
                    mSwipeRefreshLayout.setRefreshing(true);
                }
            });
        }
    }

    @Override
    public void hideLoadingIndicator() {
        Timber.d("MOXY: Refresher hideLoadingIndicator()");
        if (mSwipeRefreshLayout != null) {
            mSwipeRefreshLayout.post(() -> {
                if (mSwipeRefreshLayout != null) {
                    mSwipeRefreshLayout.setRefreshing(false);
                }
            });
        }
    }
}

File: app/src/main/kotlin/com/bankex/pay/ui/base/fragment/BaseRefresherFragment.kt
package com.bankex.pay.ui.base.fragment

import android.support.v4.widget.SwipeRefreshLayout
import kotlinx.android.synthetic.main.fr_recycler.*
import timber.log.Timber

abstract class BaseRefresherFragment : BaseNoInternetFragment(), SwipeRefreshLayout.OnRefreshListener {
    override fun onResume() {
        super.onResume()
        refresher.setOnRefreshListener(this)
    }

    override fun onPause() {
        refresher.setOnRefreshListener(null)
        super.onPause()
    }

    override fun showLoadingIndicator() {
        Timber.d("MOXY: Refresher showLoadingIndicator()")
        refresher?.post {
            refresher?.isRefreshing = true
        }
    }

    override fun hideLoadingIndicator() {
        Timber.d("MOXY: Refresher hideLoadingIndicator()")
        refresher?.post {
            refresher?.isRefreshing = false
        }
    }
}
